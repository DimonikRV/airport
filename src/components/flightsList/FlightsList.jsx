import React from 'react';
import PropTypes from 'prop-types';
import FlightsItem from '../flightsItem/FlightsItem';
import './flightslist.scss';

const FlightsList = ({ flightsData }) => {
  if (!flightsData) {
    return null;
  }
  if (!flightsData.length) {
    return <div className="search-list_empty">No Flights</div>;
  }
  return (
    <ul className="flights-table__list">
      {flightsData.map(flight => (
        <FlightsItem key={flight.id} {...flight} />
      ))}
    </ul>
  );
};

FlightsList.propTypes = {
  flightsData: PropTypes.arrayOf(PropTypes.shape()),
};

export default FlightsList;
