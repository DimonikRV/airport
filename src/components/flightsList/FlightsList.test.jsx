import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FlightsList from './FlightsList';

const flightsData = [
  {
    id: '1',
    flight: 'LX155',
    time: '12:20',
    terminal: 'B',
    way: 'Dubai',
    status: 'Scheduled',
    company: 'Swiss',
    logo: 'swiss',
  },
  {
    id: '2',
    flight: 'FR654',
    time: '13:45',
    terminal: 'A',
    way: 'Barcelona',
    status: 'Delayed',
    company: 'Ryanair',
    logo: 'ryanair',
  },
  {
    id: '3',
    flight: 'LO373',
    time: '15:10',
    terminal: 'C',
    way: 'Warsaw',
    status: 'Departed',
    company: 'LOT',
    logo: 'lot',
  },
];

describe('FlightsList', () => {
  it('renders list of flights', () => {
    render(<FlightsList flightsData={flightsData} />);

    const listItems = screen.getAllByRole('listitem');

    expect(listItems.length).toBe(flightsData.length);
  });

  it('should render "No Flights" when there are no flights data', () => {
    render(<FlightsList flightsData={[]} />);
    expect(screen.getByText('No Flights')).toBeInTheDocument();
  });

  it('list render does not render without data', () => {
    render(<FlightsList />);
    expect(screen.queryByRole('list')).toBeNull();
  });

  it('list snapshot', () => {
    const list = render(<FlightsList flightsData={flightsData} />);
    expect(list).toMatchSnapshot();
  });
});
