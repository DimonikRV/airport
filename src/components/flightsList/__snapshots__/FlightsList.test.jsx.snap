// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightsList list snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flights-table__list"
      >
        <li
          class="flights-table__item"
        >
          <div
            class="terminal-field item-slot"
          >
            B
          </div>
          <div
            class="time-field item-slot"
          />
          <div
            class="way-field item-slot"
          />
          <div
            class="status-field item-slot"
          >
            Planed at undefined
          </div>
          <div
            class="company-name item-slot"
          >
            <div
              class="logo"
            >
              <img
                alt="LOT"
              />
            </div>
            <p />
          </div>
          <div
            class="flight-field item-slot"
          />
          <div
            class="details-field item-slot"
          >
            <a
              href="/departures/2000043798526?dt=11-01-2022"
            >
              flight details
            </a>
          </div>
        </li>
        <li
          class="flights-table__item"
        >
          <div
            class="terminal-field item-slot"
          >
            A
          </div>
          <div
            class="time-field item-slot"
          />
          <div
            class="way-field item-slot"
          />
          <div
            class="status-field item-slot"
          >
            Planed at undefined
          </div>
          <div
            class="company-name item-slot"
          >
            <div
              class="logo"
            >
              <img
                alt="LOT"
              />
            </div>
            <p />
          </div>
          <div
            class="flight-field item-slot"
          />
          <div
            class="details-field item-slot"
          >
            <a
              href="/departures/2000043798526?dt=11-01-2022"
            >
              flight details
            </a>
          </div>
        </li>
        <li
          class="flights-table__item"
        >
          <div
            class="terminal-field item-slot"
          >
            C
          </div>
          <div
            class="time-field item-slot"
          />
          <div
            class="way-field item-slot"
          />
          <div
            class="status-field item-slot"
          >
            Planed at undefined
          </div>
          <div
            class="company-name item-slot"
          >
            <div
              class="logo"
            >
              <img
                alt="LOT"
              />
            </div>
            <p />
          </div>
          <div
            class="flight-field item-slot"
          />
          <div
            class="details-field item-slot"
          >
            <a
              href="/departures/2000043798526?dt=11-01-2022"
            >
              flight details
            </a>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flights-table__list"
    >
      <li
        class="flights-table__item"
      >
        <div
          class="terminal-field item-slot"
        >
          B
        </div>
        <div
          class="time-field item-slot"
        />
        <div
          class="way-field item-slot"
        />
        <div
          class="status-field item-slot"
        >
          Planed at undefined
        </div>
        <div
          class="company-name item-slot"
        >
          <div
            class="logo"
          >
            <img
              alt="LOT"
            />
          </div>
          <p />
        </div>
        <div
          class="flight-field item-slot"
        />
        <div
          class="details-field item-slot"
        >
          <a
            href="/departures/2000043798526?dt=11-01-2022"
          >
            flight details
          </a>
        </div>
      </li>
      <li
        class="flights-table__item"
      >
        <div
          class="terminal-field item-slot"
        >
          A
        </div>
        <div
          class="time-field item-slot"
        />
        <div
          class="way-field item-slot"
        />
        <div
          class="status-field item-slot"
        >
          Planed at undefined
        </div>
        <div
          class="company-name item-slot"
        >
          <div
            class="logo"
          >
            <img
              alt="LOT"
            />
          </div>
          <p />
        </div>
        <div
          class="flight-field item-slot"
        />
        <div
          class="details-field item-slot"
        >
          <a
            href="/departures/2000043798526?dt=11-01-2022"
          >
            flight details
          </a>
        </div>
      </li>
      <li
        class="flights-table__item"
      >
        <div
          class="terminal-field item-slot"
        >
          C
        </div>
        <div
          class="time-field item-slot"
        />
        <div
          class="way-field item-slot"
        />
        <div
          class="status-field item-slot"
        >
          Planed at undefined
        </div>
        <div
          class="company-name item-slot"
        >
          <div
            class="logo"
          >
            <img
              alt="LOT"
            />
          </div>
          <p />
        </div>
        <div
          class="flight-field item-slot"
        />
        <div
          class="details-field item-slot"
        >
          <a
            href="/departures/2000043798526?dt=11-01-2022"
          >
            flight details
          </a>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
